planner/core/preprocess.go

```diff
@@ -728,6 +773,60 @@ func (p *preprocessor) handleTableName(tn *ast.TableName) {
 	tn.DBInfo = dbInfo
 }
 
+func (p *preprocessor) checkNotInRepair(tn *ast.TableName) {
+	dbMap := domainutil.RepairInfo.GetTablesInRepair()
+	var dbInfo *model.DBInfo
+	for _, v := range dbMap {
+		if v.Name.L == tn.Schema.L {
+			dbInfo = v
+			break
+		}
+	}
+	if dbInfo == nil {
+		return
+	}
+	for _, t := range dbInfo.Tables {
+		if t.Name.L == tn.Name.L {
+			p.err = ddl.ErrWrongTableName.GenWithStackByArgs(tn.Name.L)
+			break
+		}
+	}
+}
+
+func (p *preprocessor) handleRepairName(tn *ast.TableName) {
+	// Check the whether the repaired table is system table.
+	if util.IsMemOrSysDB(tn.Schema.L) {
+		p.err = ddl.ErrRepairTableFail.GenWithStackByArgs("memory or System database is not for repair")
+		return
+	}
+	dbMap := domainutil.RepairInfo.GetTablesInRepair()
+	// tableName here only has the schema rather than DBInfo.
+	var dbInfo *model.DBInfo
+	for _, v := range dbMap {
+		if v.Name.L == tn.Schema.L {
+			dbInfo = v
+			break
+		}
+	}
+	if dbInfo == nil {
+		p.err = ddl.ErrRepairTableFail.GenWithStackByArgs("database " + tn.Schema.L + " is not in repair")
+		return
+	}
+	var repairTable *model.TableInfo
+	for _, t := range dbInfo.Tables {
+		if t.Name.L == tn.Name.L {
+			repairTable = t
+			break
+		}
+	}
+	if repairTable == nil {
+		p.err = ddl.ErrRepairTableFail.GenWithStackByArgs("table " + tn.Name.L + " is not in repair")
+		return
+	}
```

Comment:
We can add a return value like `dbInfo` to `GetRepairedTableInfoByTableName`, Then we can replace line802-825 with this function.

Reply:
That's a perfect catch!